most_executed_script.gs

```diff
@@ -0,0 +1,166 @@
+/**
+* Get the maximum number of executions of a cloud project
+* @param {string} cloudProjectId of the cloud project
+* @param {Date} fromTime is the date and time 
+* from which the executions has to be seen
+* @return {Object} having key as projectId and value as number of executions
+*/
+function getNumberOfExecutionOfScript(cloudProjectId, fromTime) {
+  var pageToken = null;
+  var resultData = null;
+  var max = 0;
+  var i;
+  var limit = false;
+  var processIdsMap = {};
+  var projectIdsMap = {};
+  // loop to get the first page which has enteries in it
+  do {
+    var header = {
+      'Authorization': 'Bearer ' + ScriptApp.getOAuthToken()
+    };
+    var body = {
+      'projectIds': [
+        cloudProjectId
+      ],
+      'resourceNames': [
+        'projects/' + cloudProjectId
+      ],
+      'pageToken': pageToken,
+      'orderBy': "timestamp desc"
+    };
+    var options = {
+      'method': 'post',
+      'contentType': 'application/json',
+      'headers': header,
+      'payload': JSON.stringify(body),
+      'muteHttpExceptions': false
+    };
+    var url = 'https://logging.googleapis.com/v2/entries:list';
+    var response = UrlFetchApp.fetch(url, options);
+    var json = response.getContentText();
+    resultData = JSON.parse(json);
+    pageToken = resultData.nextPageToken;
+  } while (!resultData.entries);
+
+  // loop to go over all the enteries and map the processId with the projectId
+  while (resultData.entries) {
+    for (i in resultData.entries) {
+      if (resultData.entries[i].timestamp < fromTime) {
+        limit = true;
+        break;
+      }
+
+      if (resultData.entries[i].labels) {
+        var processId =
+          resultData.entries[i].labels['script.googleapis.com/process_id'];
+        var projectId =
+          resultData.entries[i].labels['script.googleapis.com/project_key'];
+        processIdsMap[processId] = projectId;
+      }
+    }
+    if (limit) {
+      break;
+    }
+    var header = {
+      'Authorization': 'Bearer ' + ScriptApp.getOAuthToken()
+    };
+    var body = {
+      'projectIds': [
+        cloudProjectId
+      ],
+      'resourceNames': [
+        'projects/' + cloudProjectId
+      ],
+      'pageToken': pageToken,
+      'orderBy': 'timestamp desc'
+    };
+    var options = {
+      'method': 'post',
+      'contentType': 'application/json',
+      'headers': header,
+      'payload': JSON.stringify(body),
```

Comment:
this call is done previously. looks like there is some inefficiency here.

Reply:
The first call is for the first page which has entries in it as before the first page there are many blank pages that's why a do-while loop for that and the second while loop is to go through all the pages which have entries in it

Reply:
Extract that part into a helper function - "extractFirstPageOf... " or something similar.

Reply:
Okay

Reply:
Updated